### ISSUE TITLE
Provide hooks for context-like access

### ISSUE THREAD

**Issue by tyteen4a03** [MiroslavPetrik/react-form-action#7]
It would be great if the package isn't limited to renderProps but instead can let child component access state via hooks.

**Comment by MiroslavPetrik** [comment]
Yeah, I saw the `useActionState` results pushed into Context API [in the react conf by Sam Selikoff.](https://youtu.be/T8TZQ6k4SLE?t=11899)

What are your use cases for this?

**Comment by MiroslavPetrik** [comment]
I have a preliminary version which can be installed `npm i react-form-action@next`.

Here is [example usage](https://github.com/MiroslavPetrik/react-form-action/tree/context?tab=readme-ov-file#context-form)

The render props are now accessible via the `useFormContext()` hook.

There is simple & reusable [`<Pending>` component implemented.](https://github.com/MiroslavPetrik/react-form-action/blob/context/src/Pending.tsx)

Others can be defined easily:
```tsx
import { useFormContext } from "react-form-action/client";

function DebugValidationError() {
  const { isInvalid, validationError } = useFormContext();

  return isInvalid && JSON.stringify(validationError);
}
```

**Comment by MiroslavPetrik** [comment]
I've refactored the Form from createForm, so its not coupled with the context.
Having context separate out of form element enables us to read the action state outside of it.

This is good for UI styling, for example when we want to display the progress around the form, etc.

This is just a sketch, but the ActionForm eventually should be just Form.

```tsx
import { Action, ActionForm } from "react-form-action/client";

function SubscribeForm() {
  return (
      <Action action={subscribeAction} initialData={null}>
          <ActionForm>
            <input type="text" name="email" data-testid="email" />
            <button type="submit" data-testid="submit" />
          </ActionForm>
           {/* Read action state outside of <form> (ActionForm) */
          <ValidationError />
        </Action>
  );
}
```

**Comment by MiroslavPetrik** [comment]
Ok, finally the context can be used for convenient JSX components like the recently added `ZodFieldError` which takes `"errors"` and `"name"` prop. 
Now, with the `useActionContext()` the `"errors"` prop can be added internally, so users only need to specify the field name:

![image](https://github.com/user-attachments/assets/b05955fb-c964-4316-9362-94fcbc3d16bc)

![image](https://github.com/user-attachments/assets/70e4c891-9fa9-4863-822f-60e3de11895b)

**Comment by MiroslavPetrik** [comment]
released in #11
